radio/src/storage/sdcard_yaml.cpp

```diff
@@ -47,23 +49,74 @@ const char * readYamlFile(const char* fullpath, const YamlParserCalls* calls, vo
     if (result != FR_OK) {
         return SDCARD_ERROR(result);
     }
-      
+
     YamlParser yp; //TODO: move to re-usable buffer
     yp.init(calls, parser_ctx);
 
-    char buffer[32];
-    while (f_read(&file, buffer, sizeof(buffer), &bytes_read) == FR_OK) {
+    uint16_t calculated_checksum = 0xFFFF;
+    uint16_t file_checksum = 0;
 
-      // reached EOF?
-      if (bytes_read == 0)
+    bool first_block = true;
+    char buffer[32];
+    while (f_read(&file, buffer, sizeof(buffer)-1, &bytes_read) == FR_OK) {
```

Comment:
yes, because of line 62, the loop is left when there a no bytes to read. 
There is the total sum of bytes read missing and that sum has to be used to fix this.